### PR TITLE
Make sqlite PRNG context thread-local

### PR DESCRIPTION
**Pull request by nirbhayc**
Fix provided by @akshatsikarwar

### CODE REVIEW

**Comment by dorinhogea** [comment]
testcase?

File: sqlite/src/random.c [hunk 1]
```diff
@@ -21,7 +21,7 @@
 /* All threads share a single random number generator.
 ** This structure is the current state of the generator.
 */
-static SQLITE_WSD struct sqlite3PrngType {
+static __thread SQLITE_WSD struct sqlite3PrngType {
   unsigned char isInit;          /* True if initialized */
```
Comment (mistachkin) [inline sqlite/src/random.c]: To avoid future merge conflicts, can we wrap this in the normal #if blocks for SQLite customizations?

Comment (mponomar) [reply sqlite/src/random.c]: I'm assuming upstream SQLite doesn't have this bug (due to its locking vs ours?)

Comment (mistachkin) [reply sqlite/src/random.c]: Right, canonical SQLite is immune to this issue.

**Comment by nirbhayc** [comment]
Cherry-picked to 7.0.

**Pull request merged**